src/main.rs

```diff
@@ -394,7 +408,61 @@ async fn run() -> Result<()> {
         PathBuf::from(&src)
     };
 
-    let mut choices = Vec::new();
+    // Let's first find all subdirectories that are compatible with our supported recipes.
+    let root_src_dir = src_dir.clone();
+    let mut open_subdirs = vec![(0, src_dir)];
+    let mut sourceroot_candidates = vec![];
+    while !open_subdirs.is_empty() {
+        // If we call this, we have a non-empty vector.
+        let (depth, target_dir) = open_subdirs.pop().unwrap();
+        let subdirs = std::fs::read_dir(&target_dir)
+            .context(format!("failed to list subdirectories of {}", target_dir.to_string_lossy()))?;
```

Comment:
same for other uses of `context` with  `format!`
```suggestion
            .with_context(|| format!("failed to list subdirectories of {}", target_dir.display()))?;
```